src/charts/ScenarioManagerTreeList/components/SearchInput/SearchInput.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState, useRef } from 'react';
+import { TextField } from '@material-ui/core';
+import PropTypes from 'prop-types';
+
+export const SearchInput = ({ onSearchChange, className, label }) => {
+  const [textValue, setTextValue] = useState('');
+
+  const timeoutHandle = useRef();
+
+  const clearCurrentTimemout = () => {
+    if (timeoutHandle.current) {
+      clearTimeout(timeoutHandle.current);
+    }
+  };
+
+  // clear timeout if enter pressed.
+  const onKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      clearCurrentTimemout();
+      onSearchChange(textValue);
+    }
+  };
+
+  const onTextChange = (event) => {
+    const newValue = event.target.value;
+
+    setTextValue(newValue);
+    clearCurrentTimemout();
+
+    timeoutHandle.current = setTimeout(() => {
+      onSearchChange(newValue);
+    }, 500);
+  };
+
+  return (
+    <TextField
+      data-cy="scenario-manager-search-field"
+      id="standard-search"
+      variant="outlined"
+      label={label}
+      type="search"
+      className={className}
+      value={textValue}
+      onChange={onTextChange}
+      onKeyDown={onKeyDown}
+      color="primary"
+    />
+  );
+};
+
+SearchInput.propTypes = {
+  /**
+   * callback for change search value
+   */
+  onSearchChange: PropTypes.func.isRequired,
+  /**
+   * className for TextField style (optional)
+   */
+  className: PropTypes.string,
+  /**
+   * label of TextField
+   */
+  label: PropTypes.string.isRequired,
```

Comment:
`label` is defined as **required** here, and provided by `<SearchInput label={labels.searchField} ...` in _ScenarioManagerTreeList.js_, but `labels.searchField` was not required so far in `ScenarioManagerTreeList`, so it would be a breaking change for webapps that were not defining any label for the search field

Could we remove `isRequired` and allow empty labels for the search field ?

Reply:
Done